Pr/clipboard debug

uli42 opened the pull request:
lots of improvements to clipboard, mostly debugging.

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/NXproperty.c:
```diff
@@ -93,9 +93,9 @@ nxagentWMStateRec;
+int
 ProcChangeProperty(ClientPtr client)
-{	      
+{
```
Why would we want to diverge hw/nxagent/NXproperty.c and dix/proporty.c white-space wise. Is this a good idea?

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Clipboard.c:
```diff
@@ -163,6 +163,8 @@ const char * GetClientSelectionStageString(int stage)
 #endif
 
+#define SAFE_XFree(what) do {if (what) {XFree(what); what = NULL;}} while (0)
+
```
Shouldn't we define this much more globally, so that it can be reused at various other places?

sunweaver replied:
@Ionic uses this macro in his Xinerama tiling branch, too.

uli42 commented:
I have moved the macro to Utils.h, but please note that it is  a
different one from Ionic's (which calls free()  while mine calls
XFree())

You are also right about the whitespace stuff, I did not have in mind
that NXproperty.c is derived from Xorg code in dix. So I reverted that
one.
On Mon, Oct 22, 2018 at 3:26 PM Mike Gabriel <notifications@github.com> wrote:
>
> @sunweaver requested changes on this pull request.
>
> Please check my comments. Currently, Github has out-of-sync webmirrors, so commenting on code ttw is really difficult and tricky.
>
> About the FIXME regarding the reset of "i = 0" in one of the commits (Github would not let me comment inline): Not necessarily do we need to reset the "i" counter, currently it seems that the counter "i" is used to iterate over all selections, first up to nxagentCurrentSelection, then up to nxagentMaxSelections. This could be wanted. However, I am unsure and would have to investigate the clipboard code in more depths.
>
> ________________________________
>
> In nx-X11/programs/Xserver/hw/nxagent/NXproperty.c:
>
> >  ProcChangeProperty(ClientPtr client)
> -{
> +{
>
> Why would we want to diverge hw/nxagent/NXproperty.c and dix/proporty.c white-space wise. Is this a good idea?
>
> ________________________________
>
> In nx-X11/programs/Xserver/hw/nxagent/Clipboard.c:
>
> > @@ -163,6 +163,8 @@ const char * GetClientSelectionStageString(int stage)
>  #define PrintClientSelectionStage()
>  #endif
>
> +#define SAFE_XFree(what) do {if (what) {XFree(what); what = NULL;}} while (0)
> +
>
> Shouldn't we define this much more globally, so that it can be reused at various other places?
>
> ________________________________
>
> In nx-X11/programs/Xserver/hw/nxagent/Clipboard.c:
>
> > @@ -163,6 +163,8 @@ const char * GetClientSelectionStageString(int stage)
>  #define PrintClientSelectionStage()
>  #endif
>
> +#define SAFE_XFree(what) do {if (what) {XFree(what); what = NULL;}} while (0)
> +
>
> @Ionic uses this macro in his Xinerama tiling branch, too.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
